std/traits.d

```diff
@@ -6730,3 +6730,29 @@ unittest
     static assert(getSymbolsByUDA!(C, UDA)[0].stringof == "C");
     static assert(getSymbolsByUDA!(C, UDA)[1].stringof == "d");
 }
+
+/**
+   Returns: $(D true) iff all types $(D T) are the same.
+*/
+template allSameType(T...)
+{
+    static if (T.length <= 1)
+    {
+        enum bool allSameType = true;
+    }
+    else
+    {
+        enum bool allSameType = is(T[0] == T[1]) && allSameType!(T[1..$]);
+    }
+}
+
```

Comment:
BTW: @andralex there's seems to be different views (according to post in the forums) on whether variadic traits should iterate or recurse their CT-parameters. See for example my three different implementions of allSame here: 

https://github.com/nordlow/justd/blob/master/traits_ex.d#L25 (tri-divide-recursion)
https://github.com/nordlow/justd/blob/master/traits_ex.d#L47 (iteration)
https://github.com/nordlow/justd/blob/master/traits_ex.d#L79 (tail-recursion)

Which one should we strive for? Will `static foreach` enable the iterative approach to not require an inner templated function?